commands/update-version-in-file/update_version_in_file.go

```diff
@@ -0,0 +1,86 @@
+package updateversioninfile
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/kurtosis-tech/kudet/helpers"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/spf13/cobra"
+	"os"
+	"regexp"
+	"strings"
+)
+
+const (
+	updateVersionInFileCmdStr     = "update-version-in-file <to update filepath> <pattern format string> <new version>"
+	versionRegexStr               = "[0-9A-Za-z_./-]+"
+	formatStrReplacementSubstr    = "%s"
+	expectedNumSearchPatternLines = 1
+)
+
+var versionRegex = regexp.MustCompile(versionRegexStr)
+
+var UpdateVersionInFileCmd = &cobra.Command{
+	Use:   updateVersionInFileCmdStr,
+	Short: "Updates semantic version of a Kurtosis Repo in file",
+	Args:  cobra.ExactArgs(3),
+	RunE:  run,
+}
+
+func run(cmd *cobra.Command, args []string) error {
+	toUpdateFilepath, patternFormatStr, newVersion := args[0], args[1], args[2]
+	fileToUpdateInfo, err := os.Stat(toUpdateFilepath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return stacktrace.Propagate(err, "No file exists at '%s'", toUpdateFilepath)
+		}
+		return stacktrace.Propagate(err, "An error occurred attempting to retrieve file info for file at '%s'", toUpdateFilepath)
+	}
+	fileToUpdateMode := fileToUpdateInfo.Mode()
+	fileToUpdate, err := os.ReadFile(toUpdateFilepath)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred attempting to read file at '%s'", toUpdateFilepath)
+	}
+
+	if !strings.Contains(patternFormatStr, formatStrReplacementSubstr) {
```

Comment:
Under the principle of eject-early eject-often, you'd want to do this validation earlier, before you start reading the file (which in its current incarnation is a potentially slow operation). This also lets you have a flow like:

1. Validate all your axioms first
2. Start doing logic

Rather than mixing them